packages/appium/lib/extension/driver-config.js

```diff
@@ -0,0 +1,215 @@
+// @ts-check
+
+import _ from 'lodash';
+import { DRIVER_TYPE } from '../constants';
+import log from '../logger';
+import { ExtensionConfig } from './extension-config';
+
+/**
+ * @extends {ExtensionConfig<DriverType>}
+ */
+export class DriverConfig extends ExtensionConfig {
+  /**
+   * A mapping of `APPIUM_HOME` values to {@link DriverConfig} instances.
+   * Each `APPIUM_HOME` should only have one associated `DriverConfig` instance.
+   * @type {Record<string,DriverConfig>}
+   * @private
+   */
+  static _instances = {};
+
+  /**
+   * Call {@link DriverConfig.getInstance} instead.
+   * @private
+   * @param {import('./manifest').Manifest} io - IO object
+   * @param {DriverConfigOptions} [opts]
+   */
+  constructor (io, {logFn, extData} = {}) {
+    super(DRIVER_TYPE, io, logFn);
+    /** @type {Set<string>} */
+    this.knownAutomationNames = new Set();
+
+    if (extData) {
+      this.validate(extData);
+    }
+  }
+
+  /**
+   * Checks extensions for problems
+   * @param {ExtRecord<DriverType>} exts
+   */
+  validate (exts) {
+    this.knownAutomationNames.clear();
```

Comment:
is that a side effect?

Reply:
Yes.  Not my fault. 😄

Reply:
I guess we could do `this.knownAutmationNames = new Set()`

Reply:
On second thought, maybe it is my fault.  Just not _recently_ my fault. :D

So, when we validate, we're validating the entire collection of drivers found in `extensions.yaml`.

`knownAutomationNames` tracks if there's a collision in automation names, since there can only be one driver per name.

`validate()` in `ExtensionConfig` calls `getWhateverProblems()`, and _that_ function begins adding automation names to the `Set`.  

So, what we could do to avoid the side effect is to create a new `Set` in `validate()`, then explicitly pass it into each call to `getWhateverProblems()`.  This is more verbose, but probably a better idea.

Reply:
But, since this is a subclass, and `ExtensionConfig#validate` does the work, we break the abstraction by passing something driver-specific into the abstract `getConfigProblems(..)`.

The refactoring required to make this work due to the inheritance at play may not be worth the bother.

Reply:
... that being said, I think it _should_ be refactored.  Just not here.